### PR TITLE
databases-week2

### PR DESCRIPTION
**Pull request by sumanghimire79**

### CODE REVIEW

File: databases/week2/db1w2_schoolDB_DDL-DML_hwPart2.sql [hunk 1]
```diff
@@ -0,0 +1,126 @@
+-- DDL homework creating table,inserting data,deleting,updating
+CREATE DATABASE school DEFAULT CHARACTER SET = 'utf8mb4';
+USE school;
+-- creatin Class table: with the columns: id, name, begins (date), ends (date)
+CREATE TABLE `Class` (
+  `id` INT(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  `name` VARCHAR(255) NOT NULL,
+  `begin_date` DATETIME NOT NULL,
```
Comment (mani-zaeim) [inline databases/week2/db1w2_schoolDB_DDL-DML_hwPart2.sql:8]: it suppose to be begins and ends

File: databases/week2/db1w2_schoolDB_DDL-DML_hwPart2.sql [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  `email` VARCHAR(255) NOT NULL,
+  `phone` VARCHAR(255) NULL,
+  `class_id` INT(10) UNSIGNED,
+  FOREIGN KEY (`class_id`) REFERENCES `Class` (`id`) ON DELETE CASCADE ON UPDATE CASCADE
+) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4 COLLATE = utf8mb4_unicode_ci;
+-- Add a new column to the Class table named status which can only have the following VALUES: not-started, ongoing, finished (hint: enumerations).
+ALTER TABLE
+  Class
+ADD
+  status VARCHAR(255);
```
Comment (mani-zaeim) [inline databases/week2/db1w2_schoolDB_DDL-DML_hwPart2.sql:24]: Please use Enum

File: databases/week2/db1w2_schoolDB_DDL-DML_hwPart2.sql [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+ALTER TABLE
+  Class
+ADD
+  ClassRemarks VARCHAR(255);
```
Comment (mani-zaeim) [inline databases/week2/db1w2_schoolDB_DDL-DML_hwPart2.sql:28]: not required column name

File: databases/week2/db1w2_schoolDB_DDL-DML_hwPart2.sql [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+-- delete column from the table
+ALTER table
+  Class DROP COLUMN ClassRemarks;
+SELECT
```
Comment (mani-zaeim) [inline databases/week2/db1w2_schoolDB_DDL-DML_hwPart2.sql:32]: unnecessary query

File: databases/week2/db1w2hw_hyf_lesson2_hwPart1-part3.sql [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+FROM
+  task
+  join user_task ON task.id = user_task.task_id
+  JOIN user ON user_task.user_id = user.id
+WHERE
+  user.email LIKE "%@spotify.com";
+-- Get all the tasks for 'Donald Duck' with status 'Not started'
+SELECT
+  task.title,
+  user.name as user,
```
Comment (mani-zaeim) [inline databases/week2/db1w2hw_hyf_lesson2_hwPart1-part3.sql:81]: it is correct, however its good to use a better alias like user_name, the VSCode plugin show wrong result if you dont use the alias at all, but thats not the case in any other IDE or terminal, just to let you know :)

File: databases/week2/db1w2hw_hyf_lesson2_hwPart1-part3.sql [hunk 6]
```diff
@@ -0,0 +1,122 @@
+-- Part 1: Working with tasks // DDL homework
+-- Add a task with these attributes: title, description, created, updated, due_date, status_id, user_id
+INSERT INTO
```
[the remark this answers is not in the document]
Comment (sumanghimire79) [reply databases/week2/db1w2hw_hyf_lesson2_hwPart1-part3.sql:3]: i do agreed, i just tried if that works. but yes i agree with you

File: databases/week2/db1w2hw_hyf_lesson2_hwPart1-part3.sql [hunk 7]
```diff
@@ -47,12 +47,18 @@ SET
 WHERE
   id = 36;
 -- Change a task status
+  use hyf_lesson2;
 UPDATE
   task
 SET
-  status_id = 3
+  status_id = 3 -- status.name = 'Done';
+  -- should we JOIN here but how? i tried but did not work ?
```
Comment (mani-zaeim) [inline databases/week2/db1w2hw_hyf_lesson2_hwPart1-part3.sql:55]: you may use Sub Select query inorder to retrieve ID

File: databases/week2/db1w2_schoolDB_DDL-DML_hwPart2.sql [hunk 8]
```diff
@@ -8,55 +8,43 @@ CREATE TABLE `Class` (
...
-  Class
-ADD
-  ClassRemarks VARCHAR(255);
--- delete column from the table
+  Class RENAME COLUMN status1 TO status;
+-- ALTER TABLE Class ADD status VARCHAR(255);
+  -- delete column from the table
 ALTER table
-  Class DROP COLUMN ClassRemarks;
+  Class DROP COLUMN status;
```
Comment (mani-zaeim) [inline databases/week2/db1w2_schoolDB_DDL-DML_hwPart2.sql:37]: you are dropping the Status column which you added in a few line on above, are you sure you want to drop it?!

**Review by mani-zaeim — COMMENTED** [review]
Good Job

File: databases/week2/db1w2hw_animalFarm_DDL_hwPart4.sql [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+CREATE TABLE `customer` (
+  `id` INT(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  `firstName` VARCHAR(255) NOT NULL,
+  `lastName` VARCHAR(255) NOT NULL,
+  `email` VARCHAR(255) NULL,
+  `mobile` VARCHAR(10) NOT NULL,
+  `address` VARCHAR(255) NOT NULL
+) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4 COLLATE = utf8mb4_unicode_ci;
+CREATE TABLE `sales` (
+  `quantity` VARCHAR(10) NOT NULL,
```
Comment (martamatos) [inline databases/week2/db1w2hw_animalFarm_DDL_hwPart4.sql:44]: what's the difference between "quantity" here and "quantity" in the animal table?

File: databases/week2/db1w2hw_hyf_lesson2_hwPart1-part3.sql [hunk 10]
```diff
@@ -0,0 +1,128 @@
+-- Part 1: Working with tasks // DDL homework
+-- Add a task with these attributes: title, description, created, updated, due_date, status_id, user_id
+INSERT INTO
+  task
+VALUES
+  (
+    37,
```
Comment (martamatos) [inline databases/week2/db1w2hw_hyf_lesson2_hwPart1-part3.sql:7]: no need to add the id, and it's actually probably a good idea not to add it, just let MySQL auto increment it.

File: databases/week2/db1w2hw_hyf_lesson2_hwPart1-part3.sql [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+INSERT INTO
+  task (
+    id,
+    title,
+    description,
+    created,
+    updated,
+    due_date,
+    status_id
+  )
```
Comment (martamatos) [inline databases/week2/db1w2hw_hyf_lesson2_hwPart1-part3.sql:24]: shouldn't you be inserting a user_id as well?

File: databases/week2/db1w2hw_hyf_lesson2_hwPart1-part3.sql (hunk 7 above)
Comment (sumanghimire79) [reply databases/week2/db1w2hw_hyf_lesson2_hwPart1-part3.sql]: i am happy that i did it Mani, and probably i did it correctly. i had never thought in that way ! many thanks !

File: databases/week2/db1w2hw_hyf_lesson2_hwPart1-part3.sql (hunk 11 above)
Comment (martamatos) [reply databases/week2/db1w2hw_hyf_lesson2_hwPart1-part3.sql:24]: Yes, my bad, sorry :/

File: databases/week2/db1w2hw_hyf_lesson2_hwPart1-part3.sql [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+  id = 36;
+-- Change a task due date
+UPDATE
+  task
+SET
+  due_date = '2022-02-08'
+WHERE
+  id = 36;
+-- Change a task status
+UPDATE
```
[the remark this answers is not in the document]
Comment (martamatos) [reply databases/week2/db1w2hw_hyf_lesson2_hwPart1-part3.sql:51]: I suppose you could use a subquery, something like:

```
UPDATE 
task
SET
status_id = (SELECT id FROM status WHERE name="Done")
```

Check if it works plz xD

[the remark this answers is not in the document]
Comment (sumanghimire79) [reply databases/week2/db1w2hw_hyf_lesson2_hwPart1-part3.sql:51]: yes that works marta i check it . 
and i wrote this:
status_id = (
    SELECT
      status.id
    FROM
      status
    WHERE
      status.name LIKE '%Done' -- status.name = 'Done'
  )


this also works !!!

**Pull request merged**